org.testeditor.web.backend.xtext/src/main/java/org/testeditor/web/backend/xtext/index/IndexServiceClient.xtend

```diff
@@ -0,0 +1,74 @@
+package org.testeditor.web.backend.xtext.index
+
+import com.google.common.base.Predicate
+import com.google.inject.Inject
+import com.google.inject.name.Named
+import java.net.URI
+import java.util.List
+import java.util.Map
+import javax.ws.rs.client.Client
+import javax.ws.rs.client.Entity
+import javax.ws.rs.core.GenericType
+import javax.ws.rs.core.MediaType
+import org.eclipse.emf.ecore.EReference
+import org.eclipse.emf.ecore.resource.Resource
+import org.eclipse.emf.ecore.util.EcoreUtil
+import org.eclipse.xtext.resource.IEObjectDescription
+import org.eclipse.xtext.resource.XtextResource
+import org.eclipse.xtext.scoping.IGlobalScopeProvider
+import org.eclipse.xtext.scoping.IScope
+import org.eclipse.xtext.scoping.impl.SimpleScope
+
+import static extension java.util.Objects.requireNonNull
+
+class IndexServiceClient implements IGlobalScopeProvider {
+
+	val Client client
+	val URI baseURI
+
+	@Inject
+	new(@Named("index-service-client") Client client, @Named("index-service-base-URI") URI target) {
```

Comment:
how about instance var injection instead of ctor injection?

Reply:
I'd have to remove any checks performed on the arguments. Since the class only has simple null checks for now, which, one could argue, are not required when the dependencies get injected, I could just remove them.

Reply:
... I'll be needing some more convincing regarding why we should prefer field injection. I just tried to refactor this, and it makes testing considerably harder, while I don't really see any improvements regarding readability.
Maybe I am missing something, but as far as I can see, I now either have to configure DI just for testing, or break encapsulation of my class, which is even worse.

I am also not alone in thinking that field injection seems wrong (https://www.google.com/search?q=field+injection), so I'd like to put this up for debate again, if that's ok.